### PR TITLE
Add diagrams illustrating save/validation cycles.

### PR DESCRIPTION
**Pull request by markstory**
Refs cakephp/cakephp#7625

### CODE REVIEW

**Comment by antograssiot** [comment]
nice work

**Comment by antograssiot** [comment]
I've added the french translation

File: fr/orm/saving-data.rst [hunk 1]
```diff
@@ -223,6 +223,17 @@ de validation utilisé par association::
...
+Le diagramme suivant donne un aperçu de ce qui se passe à l'intérieur
+de la méthode ``newEntity()`` ou ``patchEntity()``:
+
+.. figure:: /_static/img/validation-cycle.png
+   :align: left
+   :alt: Logigramme montrant le process de conversionen entity/validation.
+
+Vous récupérerez toujours une entity en retour de ``newEntity()``. Si la
+validation échoue, votre entité contiendra des erreurs et tous les champs
+invalides seront absents l'entité créée.
```
Comment (HavokInspiration) [inline fr/orm/saving-data.rst]: absents **de**

And I'd stick with **entity** in english (there are a couple that are translated)

**Comment by HavokInspiration** [comment]
This is really nice and helpful :+1: 
Translation is good for me except for my comment.

**Comment by ravage84** [comment]
Very helpful and I guess something the newbies long wanted but couldn't spell out. :+1: 

I wonder if the same for the 2.x branch could help people migrating to 3.x...

**Comment by antograssiot** [comment]
should be good now

**Pull request merged**